lib/polo/translator.rb

```diff
@@ -65,11 +65,12 @@ def intersection(attrs, fields)
       attrs & fields.map { |pair| field_name(pair.first) }
     end
 
-    def new_field_value(field, strategy, value)
+    def new_field_value(field, strategy, instance)
+      value = instance.attributes[field]
       if strategy.nil?
         value.split("").shuffle.join
       else
-        strategy.call(value)
+        strategy.arity == 1 ? strategy.call(value) : strategy.call(value, instance)
```

Comment:
If this was a `Proc` we could just do `strategy.call(value, instance)`, unfortunately lambdas are fussy about the argument count when called.